### PR TITLE
Implement interest curve

### PR DESCRIPTION
**Pull request by liobrasil**
feat(FlowCreditMarket): implement FixedRateInterestCurve and KinkInterestCurve for variable interest rates based on utilization to enhance lending flexibility

feat(FlowCreditMarket): add support for zero-rate interest curves for testing scenarios where no interest accrues
fix(FlowCreditMarket): update event names and parameters for better clarity and consistency in the contract
test(interest_curve): add comprehensive tests for FixedRateInterestCurve and KinkInterestCurve to ensure correct interest rate calculations and edge cases
chore(tests): refactor tests to use new zero-rate curve functions and update references to ensure consistency across test cases

Closes: #???

## Description

______

For contributor use:

- [ ] Targeted PR against `master` branch
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [standards mentioned here](https://github.com/onflow/FlowALP/blob/master/CONTRIBUTING.md#styleguides).
- [ ] Updated relevant documentation 
- [ ] Re-reviewed `Files changed` in the Github PR explorer
- [ ] Added appropriate labels

### CODE REVIEW

**Review by nialexsan — APPROVED** [review]

File: cadence/tests/interest_accrual_integration_test.cdc [hunk 1]
```diff
@@ -0,0 +1,1096 @@
...
+    // Record initial state
+    let detailsBefore = getPositionDetails(pid: borrowerPid, beFailed: false)
+    let healthBefore = detailsBefore.health
+    let debtBefore = getDebitBalanceForType(details: detailsBefore, vaultType: Type<@MOET.Vault>())
+
+    // Get timestamp before time advancement
+    let timestampBefore = getBlockTimestamp()
+
+    log("=== BEFORE TIME ADVANCEMENT ===")
+    log("Block timestamp: ".concat(timestampBefore.toString()))
```
Comment (turbolent) [inline cadence/tests/interest_accrual_integration_test.cdc]: Here and below: Use string template expressions instead of `.concat`:
```suggestion
    log("Block timestamp: \(timestampBefore.toString())")
```
